binaries/runtime/src/lib.rs

```diff
@@ -303,12 +306,31 @@ async fn run(
                 if let Some(open_inputs) = open_operator_inputs.get_mut(&operator_id) {
                     open_inputs.remove(&input_id);
                     if open_inputs.is_empty() {
-                        // all inputs of the node were closed -> close its event channel
-                        tracing::trace!("all inputs of operator {}/{operator_id} were closed -> closing event channel", node.id());
-                        open_operator_inputs.remove(&operator_id);
-                        operator_channels.remove(&operator_id);
+                        // all inputs of the operator were closed
+                        if !config.wait_for_stop {
+                            // If wait_for_stop is false, close the event channel immediately
+                            tracing::trace!("all inputs of operator {}/{operator_id} were closed -> closing event channel", node.id());
+                            open_operator_inputs.remove(&operator_id);
+                            operator_channels.remove(&operator_id);
+                        } else {
+                            // If wait_for_stop is true, keep the channel open to receive stop signals
+                            // but mark that all inputs are closed
+                            tracing::trace!("all inputs of operator {}/{operator_id} were closed, but wait_for_stop is true -> keeping event channel open for stop signal", node.id());
+                            open_operator_inputs.remove(&operator_id);
+                            // Note: operator_channels is NOT removed here - the operator can still receive stop signals
```

Comment:
So if any operator is a source operator, all other operators that share the same runtime node will also wait for stop before exiting? I'm not sure if we want this behavior because this makes the operator behavior dependent on other operators.

In my opinion it would be more intuitive if we evaluate `wait_for_stop` for each operator individually. So only operators without inputs (or with explicit `wait_for_stop: true`) would wait for a stop signal. This way, nodes and operators would behave the same way.